src/main/java/io/github/genomicdatainfrastructure/discovery/datasets/application/usecases/SearchDatasetsQuery.java

```diff
@@ -56,6 +61,17 @@ public DatasetsSearchResponse execute(DatasetSearchQuery query, String accessTok
                 .build();
     }
 
+    private Map<String, Integer> collectIds(DatasetIdsCollector collector, DatasetSearchQuery query,
+            String accessToken) {
+        try {
```

Comment:
important: can you please remove this try/catch?

It is masking known and unkown issues that may come from the collectors. We should map these exceptions and retrieve to the user directly.

Reply:
The problem here is if one collector fails (e.g. Beacon), then all the collectors will also fail. Don't you think that it may be interesting for the user to see some datasets even if some others could not be retrieved?

Reply:
Okay, I see your point. Then I would expect a clear message to the user that one or multiple collectors have failed. 

I would suggest then to remove the try/catch for now and we create a story to improve the error handling properly, wdyt?

Reply:
Indeed, notifying the user is important, which is missing here. Yes let's create a story for that

Reply:
Can you remove the try/catch?